libraries/Microsoft.Bot.Builder.Ai.Translation/Translator.cs

```diff
@@ -293,15 +297,13 @@ internal AzureAuthToken(string key)
             if ((DateTime.Now - _storedTokenTime) < TokenCacheDuration)
                 return _storedTokenValue;
 
-            using (var client = new HttpClient())
             using (var request = new HttpRequestMessage())
             {
                 request.Method = HttpMethod.Post;
                 request.RequestUri = ServiceUrl;
                 request.Content = new StringContent(string.Empty);
                 request.Headers.TryAddWithoutValidation(OcpApimSubscriptionKeyHeader, this.SubscriptionKey);
-                client.Timeout = TimeSpan.FromSeconds(20);
-                var response = await client.SendAsync(request);
+                var response = await _httpClient.SendAsync(request);
```

Comment:
This should technically be wrapped in a using to ensure proper cleanup even in the face of an exception.

Reply:
I don't know that we can dispose of the client.  If someone gave it to us with custom configuration we don't know if they are using it for multiple clients.  In fact, we would recommend that they DO do that to avoid the perf impact of multiple HttpClients under load.

---
In reply to: [198877113](https://github.com/Microsoft/botbuilder-dotnet/pull/702#discussion_r198877113) [](ancestors = 198877113)

Reply:
Sorry, I mean disposing of the `response` object.